### PR TITLE
[12.0] Re-introduce module agreement_account and v10 behavior

### PR DESCRIPTION
**Pull request by alexis-via**
Cf my last comment in https://github.com/OCA/contract/issues/234

### CODE REVIEW

**Comment by alexis-via** [comment]
@max3903 This is the PR needed to take into account Akretion's requirement on the agreement module (in fact, it just re-introduces the behavior of the agreement modules in v10)

**Comment by alexis-via** [comment]
@max3903 I'll start working on OCA/contract for v13 in the next days. So I'm interested to have your opinion on this rapidly, in order to know if I continue to use these modules for v13 for my Chorus Pro modules or not.

**Comment by alexis-via** [comment]
@max3903 These changes are OK for you ?

File: agreement_sale/models/sale.py [hunk 1]
```diff
@@ -18,3 +19,8 @@ class SaleOrder(models.Model):
         ondelete="restrict",
         track_visibility='onchange', readonly=True, copy=True,
         states={'draft': [('readonly', False)], 'sent': [('readonly', False)]})
+
+    def _prepare_invoice(self):
+        vals = super(SaleOrder, self)._prepare_invoice()
```
Comment (bealdav) [inline agreement_sale/models/sale.py]:
```suggestion
        vals = super()._prepare_invoice()
```

**Review by bealdav — APPROVED** [review]
LGTM except super, no tests

**Comment by alexis-via** [comment]
@max3903 Do you have a few minutes to take a look at it before a merge ?

**Comment by bealdav** [comment]
cc @gurneyalex could you have a look please

**Review by sebastienbeau — APPROVED** [review]
LGTM code review

**Review by marcelsavegnago — APPROVED** [review]

**Comment by sebastienbeau** [comment]
Hi @pedrobaeza @max3903 

One of you can check and merge the PR please?
It's the last PR to finish the port of electronic invoicing on Odoo v12.

Thanks for your help

**Comment by pedrobaeza** [comment]
I can't say about these modules, and I would even prefer not to have them (I wanted to unify all), so better to abstain. I also don't see the relation with electronic invoicing.

**Comment by dreispt** [comment]
@max3903 Want to chime in? Any chance this can break something on your side?

**Comment by sergio-teruel** [comment]
@alexis-via Can you do a rebase to fix conflits

**Comment by alexis-via** [comment]
Let's merge ?

**Review by sergio-teruel — APPROVED** [review]

**Comment by alexis-via** [comment]
Could someone merge this please ?

**Comment by alexis-via** [comment]
Merging this would save me a lot of email work ! :) I keep receiving emails from users asking me about the Chorus Pro connector for v12... because they don't see it in the 12.0 branch, as it is blocked in a PR that depend on this one...
My mailbox would be very grateful for this :)

**Comment by pedrobaeza** [comment]
Please squash the commits and I'll merge it.

**Comment by sergio-teruel** [comment]
+1

**Comment by alexis-via** [comment]
Squashed

**Pull request merged**